Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataPatientBean.java

```diff
@@ -227,53 +421,383 @@ public void setSuffix(String suffix) {
+        this.identifierPatientID = identifierPatientID;
+    }
+
+    public String getIdentifierPatientID() {
+        return identifierPatientID;
+    }
+
+    public void setStreet1(String street1) {
+        this.street1 = street1;
+    }
+
+    public String getStreet1() {
+        return street1;
+    }
+
+    public void setStreet2(String street2) {
+        this.street2 = street2;
+    }
+
+    public String getStreet2() {
+        return street2;
+    }
+
+    public void setCity(String city) {
+        this.city = city;
+    }
+
+    public String getCity() {
+        return city;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+    }
+
+    public String getState() {
+        return state;
+    }
+
+    public void setPostal(String postal) {
+        this.postal = postal;
+    }
+
+    public String getPostal() {
+        return postal;
+    }
+
+    public void setPhonenumberValue(String phonenumberValue) {
+        this.phonenumberValue = phonenumberValue;
+    }
+
+    public String getPhonenumberValue() {
+        return phonenumberValue;
+    }
+
+    // IDs and isValidId
+    private boolean isValidPatientId() {
+        return withPatient != null && HelperUtil.isId(withPatient.getPatientId());
+    }
+
     public Long getPatientId() {
-        return updatePatientId;
+        if (isValidPatientId()) {
+            return withPatient.getPatientId();
+        }
+        return 0L;
+    }
+
+    private boolean isValidPersonnameId() {
+        return withPersonname != null && HelperUtil.isId(withPersonname.getPersonnameId());
+    }
+
+    public Long getPersonnameId() {
+        if (isValidPersonnameId()) {
+            return withPersonname.getPersonnameId();
+        }
+        return 0L;
+    }
+
+    private boolean isValidIdentifierId() {
+        return withIdentifier != null && HelperUtil.isId(withIdentifier.getIdentifierId());
+    }
+
+    public Long getIdentifierId() {
+        if (isValidIdentifierId()) {
+            return withIdentifier.getIdentifierId();
+        }
+        return 0L;
+    }
+
+    private boolean isValidAddressId() {
+        return withAddress != null && HelperUtil.isId(withAddress.getAddressId());
+    }
+
+    public Long getAddressId() {
+        if (isValidAddressId()) {
+            return withAddress.getAddressId();
+        }
+        return 0L;
+    }
+
+    private boolean isValidPhonenumberId() {
+        return withPhonenumber != null && HelperUtil.isId(withPhonenumber.getPhonenumberId());
+    }
+
+    public Long getPhonenumberId() {
+        if (isValidPhonenumberId()) {
+            return withPhonenumber.getPhonenumberId();
+        }
+        return 0L;
+    }
+
+    //read-record
+    private Phonenumber readPhonenumber(){
+        Phonenumber phonenumber;
+        if (isValidPhonenumberId()) {
+            phonenumber = loadTestDataService.getPhonenumberBy(getPhonenumberId());
```

Comment:
All the read methods can be rewritten to use single method which can accept the object type.

Reply:
there is no common interface